docs/src/references/c8y-firmware-management.md

```diff
@@ -0,0 +1,214 @@
+   the plugin send the `firmware_update` request to the child device connector by publishing an MQTT message:
+   * Topic: `tedge/$CHILD_DEVICE_ID/commands/req/firmware_update`
+   * The payload is a JSON record with the following fields
+     * `id`: A unique id generated by the plugin
+     * `name`: Name of the firmware received in the cloud request
+     * `version`:
+     * `url`: The file-transfer service entry URL(`http://$TEDGE_HTTP_ADDRESS/tedge/file-transfer/$CHILD_DEVICE_ID/firmware_update/$FILE_ID`)
+     * `sha256`: The SHA-256 checksum of the firmware file served via the `url`
+     * `attempt`: The count that indicates if this request is being resent or not
+1. On reception of the firmware update request on the topic `tedge/$CHILD_DEVICE_ID/commands/req/firmware_update`,
+   the child device connector is expected to do the following:
+   1. Send an acknowledgement of the receipt of the request by sending an executing status message via MQTT:
+      * Topic: `tedge/$CHILD_DEVICE_ID/commands/res/firmware_update`
+      * Payload must be a JSON record with the following fields
+        * `id`: The `id` of the request
+        * `status`: "executing"
+   1. `GET`s the firmware file from the `url` specified by the notification message.
+   1. Validate the integrity of the downloaded binary by matching its SHA-256 hash value
+      against the `sha256` checksum value received in the request.
+   1. Apply the downloaded firmware file update on the device using whatever device specific protocol.
+1. After applying the update, send the final operation status update to thin-edge via MQTT:
+   1. Topic: `tedge/$CHILD_DEVICE_ID/commands/res/firmware_update`
+   1. The payload must be a JSON record with the following fields:
+      * `id`: The `id` of the request received
+      * `status`: `successful` or `failed` based on the result of updating the firmware
+      * `reason`: The reason for the failure, applicable only for `failed` status.
+1. On reception of an operation status message, the plugin maps it to SmartREST and forwards it to the cloud.
+   * When a `success` or `failed` status message is finally received,
+     then the plugin cleans up the corresponding operation file at `$FIRMWARE_OP_PATH/$OP_ID` and
+     the firmware file entry in the file transfer repository at `$FILE_TRANSFER_REPO/$CHILD_DEVICE_ID/firmware_update/$FILE_ID`.
+   * If a notification message is received while none is expected,
+     i.e with an operation `id` that doesn't exist  `TEDGE_HTTP_ADDRESS_ROOT/$CHILD_DEVICE_ID/firmware_update/`,
+     then this notification message is ignored.      
+
+## Logging
+
+The plugin logs its progress and errors on to its `stderr`.
+
+* All upload and download operation requests are logged, when received and when completed,
+  with one line per file.
+* All changes to the list of managed file is logged, one line per change.
+* All errors are reported with the operation context (upload or download? which file?).
```

Comment:
What does the last line mean exactly, it seems incomplete.

Reply:
Yeah, that's the unfortunate side-effect of context switching in the middle of editing copy-pasted content ;)